namespace/bcl.adoc

```diff
@@ -0,0 +1,46 @@
+= Binary Compatibility Levels
+
+This is a classification of the levels of binary compatibility support for Java EE 8 classic applications in terms of the level of difficulty/complexity of the associated implementation and tools.
+
+== Level 0 - Jakarta EE 8 Only
+* No transformation required
+* Requires a Jakarta EE 8 container
+* No mandated support for backwards compatibility
+** Up to implementors to decide what level of support to provide
+
+== Level 1 - Jakarta EE 8 Translation to EE 9
+* Requires a direct translation of javax.* packages to jakarta.* analog
+* Requires transformation of META-INF/services to jakarta analogs
+* Requires transformation of properties files referencing javax based properties
+* Requires transformation of server specific properties that are classic javax based properties
+** This could be a specfic build or deployment step that processes an EE 8 app
+* Requires only a Jakarta EE 9 container
+* Requires the big bang javax to jakarta approach
```

Comment:
not necessarily. The translation tools could evolve with any incremental approach

Reply:
I agree, I don't see why this requires a big bang.

Reply:
Here I do see the complexity of the translation increasing if there is a mixture of javax/jakarta packages vs a straight translation to jakarta. Could say big bang is level 1 and incremental level 1.5.

Reply:
I don't see why that increases the complexity.  The *size* of the translation rules will increase as you add more javax APIs that have to be translated to jakarta APIs, but in general that wouldn't increase the complexity unless the additional javax APIs introduced previously unseen compatibility challenges.  Certainly having *fewer* javax APIs that needed to be translated would not *increase* the complexity.